torch/nn/modules/module.py

```diff
@@ -1385,3 +1432,32 @@ def _replicate_for_data_parallel(self):
         replica._is_replica = True
 
         return replica
+
+    def has_uninitialized_params(self):
+        for name, param in self.named_parameters(recurse=False):
+            if isinstance(param, _UninitializedParameter):
+                return True
+        return False
+
+    def infer_parameters(self, *args, **kwargs):
+        def initialize_hook(module, input):
+            module.initialize_parameters(*input) 
+            module._needs_initialization = False
+            if module.has_uninitialized_params():
+                raise RuntimeError('module {} has not been fully initialized'.format(module._get_name()))
+        initialize_hooks = []
+        previous_mode = self.training
+        self.train(False)
+        modules = [self]
+        while modules:
```

Comment:
You can use `self.apply()` here to apply a function to all the children from this Module